File: backend/lake-matcher/manual_override.py
import json
import os
from functools import lru_cache
from pathlib import Path
from typing import Dict, Optional


@lru_cache(maxsize=1)
def _load_overrides() -> Dict[str, Dict[str, float]]:
  override_path = os.getenv(
    "MANUAL_OVERRIDE_PATH",
    Path(__file__).resolve().parent / "manual_override.json"
  )
  try:
    with open(override_path, "r", encoding="utf-8") as handle:
      return json.load(handle)
  except FileNotFoundError:
    return {}


def manual_override_check(wdfw_name: str) -> Optional[Dict[str, float]]:
  overrides = _load_overrides()
  override = overrides.get(wdfw_name)
  if not override:
    return None
  return {
    "officialName": override["official_name"],
    "lat": override["lat"],
    "lng": override["lng"],
    "matched_score": 9007199254740991,
    "source": "manual_override"
  }

File: backend/lake-matcher/src/manualOverride.js
import fs from 'fs';
import path from 'path';
import { MANUAL_OVERRIDE_PATH } from './constants.js';

let overrides = null;
let overridePath = null;

function loadManualOverrides(customPath = MANUAL_OVERRIDE_PATH) {
  const resolvedPath = path.resolve(customPath);
  if (overrides && overridePath === resolvedPath) {
    return overrides;
  }
  if (!fs.existsSync(resolvedPath)) {
    overrides = {};
    overridePath = resolvedPath;
    return overrides;
  }
  const raw = fs.readFileSync(resolvedPath, 'utf8');
  overrides = JSON.parse(raw);
  overridePath = resolvedPath;
  return overrides;
}

export function manualOverrideCheck(wdfwName) {
  if (!wdfwName) {
    return null;
  }
  const data = loadManualOverrides();
  const override = data[wdfwName];
  if (!override) {
    return null;
  }
  return {
    officialName: override.official_name,
    lat: override.lat,
    lng: override.lng,
    matched_score: Number.MAX_SAFE_INTEGER,
    source: 'manual_override'
  };
}
